cluster/gce/gci/configure-helper.sh

```diff
@@ -0,0 +1,684 @@
+    setup-addon-manifests "addons" "registry"
+    local -r registry_pv_file="${dst_dir}/registry/registry-pv.yaml"
+    local -r registry_pvc_file="${dst_dir}/registry/registry-pvc.yaml"
+    mv "${dst_dir}/registry/registry-pv.yaml.in" "${registry_pv_file}"
+    mv "${dst_dir}/registry/registry-pvc.yaml.in" "${registry_pvc_file}"
+    # Replace the salt configurations with variable values.
+    remove-salt-config-comments "${controller_yaml}"
+    sed -i -e "s@{{ *pillar\['cluster_registry_disk_size'\] *}}@${CLUSTER_REGISTRY_DISK_SIZE}@g" "${registry_pv_file}"
+    sed -i -e "s@{{ *pillar\['cluster_registry_disk_size'\] *}}@${CLUSTER_REGISTRY_DISK_SIZE}@g" "${registry_pvc_file}"
+    sed -i -e "s@{{ *pillar\['cluster_registry_disk_name'\] *}}@${CLUSTER_REGISTRY_DISK}@g" "${registry_pvc_file}"
+  fi
+  if [[ "${ENABLE_NODE_LOGGING:-}" == "true" ]] && \
+     [[ "${LOGGING_DESTINATION:-}" == "elasticsearch" ]] && \
+     [[ "${ENABLE_CLUSTER_LOGGING:-}" == "true" ]]; then
+    setup-addon-manifests "addons" "fluentd-elasticsearch"
+  fi
+  if [[ "${ENABLE_CLUSTER_UI:-}" == "true" ]]; then
+    setup-addon-manifests "addons" "dashboard"
+  fi
+  if echo "${ADMISSION_CONTROL:-}" | grep -q "LimitRanger"; then
+    setup-addon-manifests "admission-controls" "limit-range"
+  fi
+
+  # Place addon manager pod manifest
+  cp "${src_dir}/kube-addon-manager.yaml" /etc/kubernetes/manifests
+}
+
+# Starts a fluentd static pod for logging.
+function start-fluentd {
+  echo "Start fluentd pod"
+  if [[ "${ENABLE_NODE_LOGGING:-}" == "true" ]]; then
+    if [[ "${LOGGING_DESTINATION:-}" == "gcp" ]]; then
+      cp "${KUBE_HOME}/kube-manifests/kubernetes/fluentd-gcp.yaml" /etc/kubernetes/manifests/
+    elif [[ "${LOGGING_DESTINATION:-}" == "elasticsearch" ]]; then
+      cp "${KUBE_HOME}/kube-manifests/kubernetes/fluentd-es.yaml" /etc/kubernetes/manifests/
+    fi
+  fi
+}
+
+
+########### Main Function ###########
+echo "Start to configure instance for kubernetes"
+
+KUBE_HOME="/home/kubernetes"
+if [[ ! -e "${KUBE_HOME}/kube-env" ]]; then
+  echo "The ${KUBE_HOME}/kube-env file does not exist!! Terminate cluster initialization."
+  exit 1
+fi
+
+source "${KUBE_HOME}/kube-env"
+config-ip-firewall
+create-dirs
+if [[ "${KUBERNETES_MASTER:-}" == "true" ]]; then
+  mount-master-pd
+  create-master-auth
+  create-master-kubelet-auth
+else
+  create-kubelet-kubeconfig
+  create-kubeproxy-kubeconfig
+fi
+
+assemble-docker-flags
+load-docker-images
+start-kubelet
+
+if [[ "${KUBERNETES_MASTER:-}" == "true" ]]; then
+  compute-master-manifest-variables
+  start-etcd-servers
+  start-kube-apiserver
+  start-kube-controller-manager
+  start-kube-scheduler
+  start-kube-addons
+else
+  start-kube-proxy
+  # Kube-registry-proxy
+  if [[ "${ENABLE_CLUSTER_REGISTRY:-}" == "true" ]]; then
+    cp "${KUBE_HOME}/kube-manifests/kubernetes/kube-registry-proxy.yaml" /etc/kubernetes/manifests
+	fi
+fi
+start-fluentd
+echo "Done for the configuration for kubernetes"
```

Comment:
The "licensing work" means https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/configure-vm.sh#l104-116?